test/cpp/interop/interop_client.h

```diff
@@ -39,11 +39,17 @@ typedef std::function<std::shared_ptr<Channel>(void)> ChannelCreationFunc;
 
 class InteropClient {
  public:
+  using LoadReportsFromLbPolicy =
+      std::shared_ptr<std::vector<xds::data::orca::v3::OrcaLoadReport>>;
+
+  static LoadReportsFromLbPolicy InitializeCustomLbPolicy();
```

Comment:
Instead of exposing this type and passing it into the ctor, why not just register the LB policy inside the ctor and have the callback directly write to the data member of `InteropClient`, so that it doesn't need to be a `shared_ptr<>`?  Even if you need this behavior to be configurable for the Android client, you can pass in a bool indicating whether to configure the LB policy.

Reply:
`serviceStub_` creation opens a channel, so LB policy needs to be created before that one is created. I considered multiple approaches but then decided that passing it here will offer more flexibility as I am not familiar enough with all the tests that use this code.

Reply:
Please take a look at a new implementation

Reply:
Thanks, I like this a lot better, since the ugliness is now hidden inside the client code.

In the long run, I think we should probably change this LB policy to get the address of the callback from a channel arg instead of it being globally set at LB policy registration time.  But that doesn't need to block this PR.

Reply:
Done